src/DataTable/utils/getVisibleColumns.jsx

```diff
@@ -35,12 +37,22 @@ export const selectColumn = {
   // Proptypes disabled as this prop is passed in separately
   /* eslint-disable react/prop-types */
   Cell: ({ row }) => {
+    const {
+      isSelectable, maxSelectedRows, onMaxSelectedRows, state: { selectedRowIds },
+    } = useContext(DataTableContext);
     const updatedProps = useConvertIndeterminateProp(row.getToggleRowSelectedProps());
+    const { index } = row;
+    const isRowSelected = index in selectedRowIds;
+    const selectedRowsLength = Object.keys(selectedRowIds).length;
+    const hasMaxSelectedRows = maxSelectedRows && maxSelectedRows === selectedRowsLength;
```

Comment:
[sanity check] Does this conditional logic handle when `maxSelectedRows === 0`?

Reply:
It doesn't because if `maxSelectedRows` is equal to `0` in this case would be `false`, should we handle it? It would be odd that the rows are selectable but I can not select one of them 😅 don't you think? I'm wondering if that would be a case use

Reply:
If `maxSelectedRows` is `0`, `hasMaxSelectedRows` would be `0`, not `false` (which goes against the implied boolean by the variable name).

> should we handle it? It would be odd that the rows are selectable but I can not select one of them 😅 don't you think? I'm wondering if that would be a case use

You're right that it's unlikely to be a use case. However, I still feel the component should adhere to the `0` if passed by the consumer. Otherwise, it's not really "correct" (the consumer is telling the component `0` rows should be selected lol). Allowing selections with `maxSelectedRows=0` doesn't feel great imho 😄 What if that `0` is intentional?